libcloud/compute/base.py

```diff
@@ -1316,7 +1320,7 @@ def filter_addresses(addresses):
         uuids = set([node.uuid for node in nodes])
 
         while time.time() < end:
-            all_nodes = self.list_nodes()
+            all_nodes = self.list_nodes(ex_cloud_opts)
```

Comment:
This is good solution. I have implemented it and edited the docstring to describe the changes.